archive/imo/imo1969_q1.lean

```diff
@@ -19,7 +19,7 @@ open int nat
 
 /-- `good_nats` is the set of natural numbers satisfying the condition in the problem
 statement, namely the `a : ℕ` such that `n^4 + a` is not prime for any `n : ℕ`. -/
-def good_nats : set ℕ := {a : ℕ | ∀ n : ℕ, ¬ prime (n^4 + a)}
+def good_nats : set ℕ := {a : ℕ | ∀ n : ℕ, ¬ nat.prime (n^4 + a)}
```

Comment:
Why do you suddenly need to add the `nat.` prefix all over mathlib? Did some change you made affect elaboration?

Reply:
These are filed that didn't know `_root_.prime` existed, before adding the `euclidean_domain` → `gcd_monoid.basic` import; now that they do, `prime` is ambiguous in the face of `open nat`.

Reply:
Oh, I forgot to mention: I could also revert this, and put `euclidean_domain.gcd_monoid` in a separate file, if that's better.

Reply:
I'm not actually sure why we don't just use `_root_.prime` throughout for `nat.prime`. So making `nat.prime` less convenient to use is no problem for me.